FFmpegInterop/UncompressedAudioSampleProvider.cpp

```diff
@@ -17,161 +17,152 @@
-
-// Minimum duration for uncompressed audio samples (200 ms)
-const LONGLONG MINAUDIOSAMPLEDURATION = 2000000;
-
-UncompressedAudioSampleProvider::UncompressedAudioSampleProvider(
-	FFmpegReader^ reader,
-	AVFormatContext* avFormatCtx,
-	AVCodecContext* avCodecCtx)
-	: UncompressedSampleProvider(reader, avFormatCtx, avCodecCtx)
-	, m_pSwrCtx(nullptr)
-{
-}
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Media::MediaProperties;
+using namespace winrt::Windows::Storage::Streams;
+using namespace std;
 
-HRESULT UncompressedAudioSampleProvider::AllocateResources()
+namespace winrt::FFmpegInterop::implementation
 {
-	HRESULT hr = S_OK;
-	hr = UncompressedSampleProvider::AllocateResources();
-	if (SUCCEEDED(hr))
+	UncompressedAudioSampleProvider::UncompressedAudioSampleProvider(_In_ AVStream* stream, _In_ Reader& reader) :
+		UncompressedSampleProvider(stream, reader),
+		m_minAudioSampleDur(ConvertToAVTime(MIN_AUDIO_SAMPLE_DUR_MS, MS_PER_SEC, m_stream->time_base))
 	{
-		// Set default channel layout when the value is unknown (0)
-		int64 inChannelLayout = m_pAvCodecCtx->channel_layout ? m_pAvCodecCtx->channel_layout : av_get_default_channel_layout(m_pAvCodecCtx->channels);
-		int64 outChannelLayout = av_get_default_channel_layout(m_pAvCodecCtx->channels);
-
-		// Set up resampler to convert any PCM format (e.g. AV_SAMPLE_FMT_FLTP) to AV_SAMPLE_FMT_S16 PCM format that is expected by Media Element.
-		// Additional logic can be added to avoid resampling PCM data that is already in AV_SAMPLE_FMT_S16_PCM.
-		m_pSwrCtx = swr_alloc_set_opts(
-			NULL,
-			outChannelLayout,
-			AV_SAMPLE_FMT_S16,
-			m_pAvCodecCtx->sample_rate,
-			inChannelLayout,
-			m_pAvCodecCtx->sample_fmt,
-			m_pAvCodecCtx->sample_rate,
-			0,
-			NULL);
-
-		if (!m_pSwrCtx)
+		if (m_codecContext->sample_fmt != AV_SAMPLE_FMT_S16)
 		{
-			hr = E_OUTOFMEMORY;
+			// Set up resampler to convert to AV_SAMPLE_FMT_S16 PCM format
+			int64_t inChannelLayout{ m_codecContext->channel_layout != 0 ? static_cast<int64_t>(m_codecContext->channel_layout) : av_get_default_channel_layout(m_codecContext->channels) };
+			int64_t outChannelLayout{ av_get_default_channel_layout(m_codecContext->channels) };
+
+			m_swrContext.reset(swr_alloc_set_opts(
+				m_swrContext.release(),
+				outChannelLayout,
+				AV_SAMPLE_FMT_S16,
+				m_codecContext->sample_rate,
+				inChannelLayout,
+				m_codecContext->sample_fmt,
+				m_codecContext->sample_rate,
+				0,
+				nullptr));
+			THROW_IF_NULL_ALLOC(m_swrContext);
+			THROW_HR_IF_FFMPEG_FAILED(swr_init(m_swrContext.get()));
 		}
 	}
 
-	if (SUCCEEDED(hr))
+	void UncompressedAudioSampleProvider::SetEncodingProperties(_Inout_ const IMediaEncodingProperties& encProp, _In_ bool setFormatUserData)
 	{
-		if (swr_init(m_pSwrCtx) < 0)
-		{
-			hr = E_FAIL;
-		}
+		// We intentionally don't call SampleProvider::SetEncodingProperties() here as
+		// it would set encoding properties with values for the compressed audio type.
```

Comment:
>values for the compressed audio type. [](start = 43, length = 37)

Shouldn't we be setting the appropriate properties for the uncompressed type here (sample rate, channel count, bitrate and bits per sample)?